accepted/2022/FunctionPointers.md

```diff
@@ -0,0 +1,706 @@
+{
+    public static bool VoidStar(void* doit, int v) => ((delegate*<int, bool>) doit)(v);
+    public static bool IntPtr(IntPtr doit, int v) => ((delegate*<int, bool>) doit)(v);
+    public static bool DelegateStar(delegate*<int, bool>* doit, int v) => (*doit)(v);
+    public static bool DelegateStarMismatch(delegate*<string, bool>* doit, int v) => (*doit)("");
+}
+```
+
+There is no proposal to extend `Pointer` for use with function pointers, and as shown above a pointer to a function pointer can be used to support strongly-typed, validated function pointers.
+
+There is also no proposal to add an analogous `FunctionPointer` type to contain both the type and address which would allow passing a function pointer directly without using `IntPtr`. This is considered a lower-priority feature and can be added later but would require all runtimes to add special behavior for this.
+
+For [the proposed byref reflection APIs](https://github.com/dotnet/runtime/issues/45152) to be added in V8, which do not require boxing and perform little to no marshalling, reflection would allow a strongly-typed, validated function pointer parameter without requiring `Pointer\FunctionPointer` since it performs simple type equality checks against the caller's and callee's parameter types.
+
+## Overidden members for a function pointer type
+Below are the non-trivial behaviors overridden by a function pointer type.
+
+### `ToString()`
+Returns a friendly string in the format `return_type + "(" + optional_parameter_types_comma_separated + ")"` and compose with the `ToString()` of other types:
+ToString() result | Delegate signature
+---|---
+"System.Void()" | `delegate*<void>`
+"System.Int32()" | `delegate*<int>`
+"System.Int32(System.String)" | `delegate*<string, int>`
+"System.Int32(System.String, System.Boolean*&)" | `delegate*<string, ref bool*, int>`
+"System.Int32()*" | `delegate*<int>*`
+"System.Int32()[]" | `delegate*<int>[]`
+"System.Int32()*[]" | `delegate*<int>*[]`
+"System.Int32()()" | `delegate*<delegate*<int>>`
+"System.Boolean(System.String(System.Int32))" | `delegate*<delegate*<int, string>, bool>`
+
+### `FullName` and `AssemblyQualifiedName` properties
+Returns `null`. This is a simplification that avoids a [type name grammar update](https://docs.microsoft.com/dotnet/framework/reflection-and-codedom/specifying-fully-qualified-type-names) that would need to include grammar for custom modifiers as well as supporting  `Type.GetType()`, `Assembly.GetType()` and any other similar APIs that parse those strings in order to construct a function pointer from it.
+
+Support for a grammar update is possible in the future, with a minimal breaking change here.
+
+Note that this property is already nullable and does return `null` in open generic parameters cases as well.
+
+### `Name` property
+Returns `String.Empty()`. Any other syntax here is somewhat arbitrary, and since the property is not nullable, the empty string seems fine unless a compelling reason is found. 
```

Comment:
```suggestion
Returns `string.Empty`. Any other syntax here is somewhat arbitrary, and since the property is not nullable, the empty string seems fine unless a compelling reason is found. 
```